src/feedback_forensics/app/data/handler.py

```diff
@@ -0,0 +1,447 @@
+    the final output plot. All annotators are columns in the original votes_df.
+
+    This also gives warning if not all annotators are available across all datasets.
+    """
+    # get mappings from visible names to column names for each dataset
+    visible_to_cols = {}
+    for dataset_name, votes_dict in votes_dicts.items():
+        metadata = votes_dict["annotator_metadata"]
+        visible_to_col = {
+            value["annotator_visible_name"]: col for col, value in metadata.items()
+        }
+        visible_to_cols[dataset_name] = visible_to_col
+
+    updated_annotator_visible_names = []
+    for annotator_name in annotator_visible_names:
+        add_annotator = True
+        for dataset_name, visible_to_col in visible_to_cols.items():
+            if annotator_name not in visible_to_col:
+                logger.warning(
+                    f"Annotator '{annotator_name}' (visible name) not found in dataset '{dataset_name}'. Skipping this annotator. Available annotators in this dataset: {list(visible_to_col.keys())}"
+                )
+                add_annotator = False
+        if add_annotator:
+            updated_annotator_visible_names.append(annotator_name)
+
+    # get column names for the remaining annotators
+    col_names = [
+        visible_to_col[annotator_name]
+        for annotator_name in updated_annotator_visible_names
+    ]
+    return col_names
+
+
+def _split_votes_dict(
+    votes_dict: dict,
+    split_col: str,
+    selected_vals: list[str] | None = None,
+) -> dict:
+    """Split votes data by split_col.
+
+    First assert that only one votes_df in vote_dfs, and split that votes_df into multiple, based on the unique values of split_col.
+
+    Args:
+        votes_dict: Dictionary with keys "df" and "annotator_metadata"
+        split_col: Column to split on
+        selected_vals: Optional list of values to filter split_col by. If None, use all values.
+
+    Returns:
+        Dictionary mapping split values to filtered DataFrames
+    """
+
+    votes_df = votes_dict["df"]
+    votes_df[split_col] = votes_df[split_col].astype(str)
+
+    if selected_vals:
+        # Filter to only selected values before grouping
+        votes_df = votes_df[votes_df[split_col].isin(selected_vals)]
+
+    grouped_df = votes_df.groupby(split_col)
+
+    split_dicts = {}
+    for name, group in grouped_df:
+        split_dicts[name] = {
+            "df": group,
+            "annotator_metadata": votes_dict["annotator_metadata"],
+            "reference_annotator_col": votes_dict["reference_annotator_col"],
+            "shown_annotator_rows": votes_dict["shown_annotator_rows"],
+        }
+
+    return split_dicts
+
+
+class ColumnHandler:
+    """Class to handle data operations (loading, computing metrics) of a single annotation column dataset."""
+
+    def __init__(self, cache: dict | None = None, avail_datasets: dict | None = None):
+
+        self._cache = cache
```

Comment:
I'd prefer to split logic further from loading and caching. Would be great to have UI, loading, and logic separated. (Just a thought, not for this PR)